syntax = "proto3";

service OpenCV {
    rpc RunOpenCV (OpenCVRequest) returns (OpenCVResponse) {}
}

message OpenCVRequest {
    string business_id          = 1;
    int32 interface_mode        = 2;
    bytes image_buffer          = 3;
    string image_url            = 4;
    string trace_id             = 5;
    int32 request_type          = 6;
    int32 context_id            = 7;
}

message OpenCVResponse {
    message Category {
        float fstd                  = 1;
        float fscore                = 2;
    }
    repeated Category results   = 1;
    string trace_id             = 2;
    int32 context_id            = 3;
    int32 error_code            = 4;
}